src/api/libcellml/componententity.h

```diff
@@ -146,35 +143,33 @@ class LIBCELLML_EXPORT ComponentEntity: public ImportedEntity
      * @brief Take the units at the given @p index and return it.
      *
      * Removes the units at the given @p index position and returns it.
-     * If an invalid @p index is passed to the method a @c std::out_of_range
-     * exception is thrown.
+     * @p index must be in the range [0, #units).
      *
      * @param index The index of the units to take (zero-based).
      *
-     * @return The units at the given @p index.
+     * @return A reference to the units at the given @p index if the index is valid, @c nullptr otherwise.
      */
     UnitsPtr takeUnits(size_t index);
 
     /**
      * @brief Take the units with the given @p name and return it.
      *
-     * Takes the units with the given @p name and returns it.
-     * If an invalid @p name is passed to the method a @c std::out_of_range
-     * exception is thrown.
+     * Takes the first occurence of the units with the given name @p name and returns it.
+     * If no units with name @p name is found then a @c nullptr is returned.
      *
      * @overload
      *
      * @param name The name of the units to take.
      *
-     * @return The Units identified with the given @p name.
+     * @return A reference to the units identified with the given @p name if found, @c nullptr otherwise
      */
     UnitsPtr takeUnits(const std::string &name);
 
     /**
      * @brief Replace a units at the given @p index.
      *
-     * Replaces the units at the given @p index with @p units.  If an invalid @p index
-     * is passed to the method a @c std::out_of_range exception is thrown.
+     * Replaces the units at the given @p index with @p units.
+     * @p index must be in the range [0, #units).
```

Comment:
Minor: why not mention that information at line 174? (Same elsewhere in the code.)